### ISSUE TITLE
No symbols found - Visual Studio Code without internet

### ISSUE THREAD

**Issue by MahadevanSrinivasan** [microsoft/vscode-cpptools#1047]
* Windows 7
    * VS Code version 1.16.1 x64
    * C/C++ extension version 0.12.4
    * step-by-step instructions to reproduce the issue
      Launch Visual Studio Code
      Add a folder with the source files (C/C++/H files)
      Open any source file and hit Ctrl+Shift+O and notice "No symbols found"
      Trying "Go to definition" or  "Peak Definition" leads to "No symbol found"
      Hovering over symbol shows "Loading" and nothing else
      Microsoft.VSCode.CPP.Extension.Exe *32 does not show any activity no matter what I do (Reset Intellisense database, Go to definition etc.,)
     Problems window shows: "No problems have been detected in the workspace so far"
 c_cpp_properties.json (stripped)
  ```
 {
            "name": "Win32",
            "includePath": [
                "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include",
                "${workspaceRoot}"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE"
            ],
            "intelliSenseMode": "msvc-x64",
            "browse": {
                "path": [
                    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/include/*",
                    "${workspaceRoot}"
                ],
                "limitSymbolsToIncludedHeaders": true,
                "databaseFilename": ""
            }
```
settings.json

```
{
    "git.ignoreLegacyWarning": true,
    "workbench.colorTheme": "Visual Studio Light",
    "window.zoomLevel": 0,
    "editor.minimap.enabled": false,
    "editor.renderWhitespace": "none",
    "C_Cpp.intelliSenseEngine": "Tag Parser",
    "workbench.activityBar.visible": true,
    "C_Cpp.errorSquiggles": "Enabled",
    "C_Cpp.intelliSenseEngineFallback": "Enabled"
}
```

**Comment by sean-mcmanus** [comment]
What happens when you change the databaseFilename property to something like "${workspaceRoot}/vc.db"? What size is the file? Can you open the folder without opening any files and see if the process uses CPU and there's a flame in the bottom right? Opening larger folders should trigger more CPU usage and a larger database. Has our extension worked for you in the past?

**Comment by MahadevanSrinivasan** [comment]
Changed the databaseFilename property to "${workspaceRoot}/vc.db" and restarted Code. Don't see the db file where it is supposed to be. 
Also tried opening the folder without opening any files - no change in the process activity. No flame at the bottom right. Just a smiley symbol. 
Extension works on a machine with access to the internet. It has not worked on this machine (without internet) before. This is the first time I am trying it. I installed the extension manually by downloading the VSIX package from the market place. Are there any dependencies I need to install manually?

**Comment by sean-mcmanus** [comment]
Yeah, we download OS dependencies on first run. But I don't know where you got the Microsoft.VSCode.CPP.Extension.exe process, because that should not be in the .vsix, but part of the download. You should install the .vsix on a machine with internet access and then zip/copy the entire folder after the dependencies are downloaded, assuming both machines have the same OS.

**Comment by MahadevanSrinivasan** [comment]
I got VSIXPackage from here: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools using instructions from here: https://stackoverflow.com/questions/37071388/how-to-install-vscode-extensions-offline.

I did as you suggested - installed Visual Studio Code on another Windows 7 machine, installed the extension, made sure the dependencies got installed, verified the symbol search and copied the entire "C:\Program Files\Microsoft VS Code\resources" to my offline machine. "Offline machine" didn't realize the CPP extension was installed. 

So, I then tried to copy the entire "Microsoft VS Code" folder. Still, the offline machine is trying to contact the server. I see the following messages in the Output window:
```

Downloading package 'C/C++ language components (Windows)'  Failed. Retrying... Failed. Retrying... Failed. Retrying...Waiting 8 seconds... Failed. Retrying...Waiting 16 seconds... Failed. Retrying...Waiting 32 seconds...
Failed at stage: downloadPackages
Error: connect EACCES 104.91.221.69:443
	at Object.exports._errnoException (util.js:1050:11)
	at exports._exceptionWithHostPort (util.js:1073:20)
	at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1097:14)

Finished
```

**Comment by sean-mcmanus** [comment]
Our extension gets installed to `C:\Users\user\.vscode\extensions\ms-vscode.cpptools-0.12.4`. You should copy/paste that folder.

**Comment by MahadevanSrinivasan** [comment]
Awesome. That did it. It will be great if this offline installation procedure is documented somewhere.

**Comment by mertzjames** [comment]
There's a stretch goal for an offline installer in this issue

https://github.com/Microsoft/vscode-cpptools/issues/298

**Comment by bobbrow** [comment]
@KronoSKoderS we completed that task and will begin releasing "offline" packages starting with our December release.  We did a test run in issue #19 which seemed to work out for those who tried it (linux-only).

**Comment by sean-mcmanus** [comment]
Fixed with 0.14.4. See the offline installer links at https://github.com/Microsoft/vscode-cpptools/releases .
